### PR TITLE
Update IT advisories

### PR DESCRIPTION
**Pull request by justmurphy**
#  #

## ✅ Pre-approval checklist ##

- [ ] This PR has an informative and human-readable title.
- [ ] Changes are limited to a single goal - *eschew scope creep!*
- [ ] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [ ] All relevant type-of-change labels have been added.
- [ ] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [ ] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [ ] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Review by mstrad — CHANGES_REQUESTED** [review]
The CSAF itself looks good. However the hash file and signature file of the CSAF should be regenerated to reflect the changed CSAF.

File: csaf_files/IT/white/2024/va-24-201-01.json [hunk 1]
```diff
@@ -94,7 +94,7 @@
             "branches": [
               {
                 "category": "product_version_range",
-                "name": "all versions",
+                "name": "<=4.8.1",
```
Comment (justmurphy) [inline csaf_files/IT/white/2024/va-24-201-01.json:97]: https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#31231-branches-type---name-under-product-version

"all versions" is invalid. See documentation above.

File: csaf_files/IT/white/2024/va-24-201-01.json [hunk 2]
```diff
@@ -158,7 +158,7 @@
                   "product_id": "CSAFPID-0078"
                 },
                 "category": "product_version_range",
-                "name": "all versions"
+                "name": "<4.8.4"
```
Comment (justmurphy) [inline csaf_files/IT/white/2024/va-24-201-01.json:161]: See above comment re: version:
https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#31231-branches-type---name-under-product-version

File: csaf_files/IT/white/2024/va-24-201-01.json [hunk 3]
```diff
@@ -229,8 +229,9 @@
       "acknowledgments": [
         {
           "names": [
-            "Matthew Galligan, CISA Rapid Action Force"
-          ]
+            "Matthew Galligan"
+          ],
+          "organization": "CISA Rapid Action Force",
```
Comment (justmurphy) [inline csaf_files/IT/white/2024/va-24-201-01.json:234]: https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#311-acknowledgments-type

Properties for acknowledgements are names, organization, summary, and urls. Moved organization from names property to organization property.

File: csaf_files/IT/white/2024/va-24-201-01.json [hunk 4]
```diff
@@ -319,8 +320,9 @@
...
         }
```
Comment (justmurphy) [inline csaf_files/IT/white/2024/va-24-201-01.json:326]: See above comment:
https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#311-acknowledgments-type

File: csaf_files/IT/white/2024/va-24-201-01.json [hunk 5]
```diff
@@ -409,8 +411,9 @@
...
```
Comment (justmurphy) [inline csaf_files/IT/white/2024/va-24-201-01.json:417]: See above comment:
https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#311-acknowledgments-type

**Comment by justmurphy** [comment]
> The CSAF itself looks good. However the hash file and signature file of the CSAF should be regenerated to reflect the changed CSAF.

@mstrad the issue I am seeing is the product tree structure: `vendor` -> `product_family` -> `product_name` -> `product_version`. 

Going to revise the advisory to meet recommended structure and will ping you for another review. I believe that all the IT advisories will need to be revised for product tree structure, and then small edits to the `product_version_range` and `acknowledgement` properties.

File: csaf_files/IT/white/2024/va-24-201-01.json (hunk 1 above)
Comment (mstrad) [reply csaf_files/IT/white/2024/va-24-201-01.json:97]: I agree with the above change.

File: csaf_files/IT/white/2024/va-24-201-01.json (hunk 2 above)
Comment (mstrad) [reply csaf_files/IT/white/2024/va-24-201-01.json:161]: I agree with the above change.

File: csaf_files/IT/white/2024/va-24-201-01.json (hunk 3 above)
Comment (mstrad) [reply csaf_files/IT/white/2024/va-24-201-01.json:234]: I agree with the above change.

File: csaf_files/IT/white/2024/va-24-201-01.json (hunk 4 above)
Comment (mstrad) [reply csaf_files/IT/white/2024/va-24-201-01.json:326]: I agree with the above change.

File: csaf_files/IT/white/2024/va-24-201-01.json (hunk 5 above)
Comment (mstrad) [reply csaf_files/IT/white/2024/va-24-201-01.json:417]: I agree with the above change.

**Comment by mstrad** [comment]
> > The CSAF itself looks good. However the hash file and signature file of the CSAF should be regenerated to reflect the changed CSAF.
> 
> @mstrad the issue I am seeing is the product tree structure: `vendor` -> `product_family` -> `product_name` -> `product_version`.
> 
> Going to revise the advisory to meet recommended structure and will ping you for another review. I believe that all the IT advisories will need to be revised for product tree structure, and then small edits to the `product_version_range` and `acknowledgement` properties.

@justmurphy I see what you're saying. Initially I was only reviewing the proposed changes in this PR itself. All of which I agree with and offer the reminder that new hashes and signature files will need to be made and pushed to the repo when a CSAF file is updated. :)

To the comment above, I agree. I took a look at the raw CSAF file and I agree that following the recommended product tree structure should be followed with the ending leaf being a product_version or product_version_range. Will this recommendation become a requirement in future CSAF versions?

File: csaf_files/IT/white/2024/va-24-201-01.json (hunk 1 above)
Comment (amanion-cisa) [reply csaf_files/IT/white/2024/va-24-201-01.json:97]: IIUC Adminer "4.8.1" happens to be a "Vers -like Specifier (vls)" so I think this change works syntactically and I understand that "all versions" is not valid.

However I'm slightly concerned with how to interpret this limit on the version range. The product is used in status and remediation elements. "<=4.8.1" might imply that there might be a more recent version, and is that more recent version fixed? We can't know the future at the time of publication, could update the CSAF if a new/fixed version is released, and the CSAF VEX profile has an explicit "Fixed" status, but at least for those familar with the CVE ecosystem, "<=4.8.1" could imply "fixed in >4.8.1, now go look to see if any such versions exist."

**Review by amanion-cisa — APPROVED** [review]
FWIW I'm fine with the changes, they are fixing technically incorrect CSAF. I can make the updates including new hashes, signatures, and ROLIE files if you'd like.

File: csaf_files/IT/white/2024/va-24-201-01.json (hunk 1 above)
Comment (justmurphy) [reply csaf_files/IT/white/2024/va-24-201-01.json:97]: @amanion-cisa there is also this as an option:
Version Range Specifier (vers)
vers is an ongoing community effort to address the problem of version ranges. Its draft specification is available at [VERS].
vers MUST be used in its canonical form. To convey the term "all versions" the special string vers:all/* MUST be used.

Comment (amanion-cisa) [reply csaf_files/IT/white/2024/va-24-201-01.json:97]: I used vers:all/* for the Adminer advisory, I think this is more correct informationally, see https://github.com/cisagov/CSAF/compare/develop...amanion-cisa:CSAF:jmfixes#diff-b68d7bbabc197e783ddff83a11b37d5c9e309aeec18c89cf5551b478586daac1L97-R166.

**Comment by amanion-cisa** [comment]
Here are fixes for version ranges and acknowledgments (and section headings, not initially part of this PR): https://github.com/cisagov/CSAF/compare/develop...amanion-cisa:CSAF:jmfixes

I won't send a PR until we've reviewed these changes.

**Comment by mstrad** [comment]
I noticed that the ROLIE feed has a small typo in the links array of each entry. Each url listed should have "href" and "rel" instead of "ref"

image

**Comment by justmurphy** [comment]
Resolved by #138

**Pull request closed without merging**